sktime/forecasting/reconcile.py

```diff
@@ -0,0 +1,355 @@
+            y = self._add_totals(y)
+
+        if X is not None:
+            if _check_index_no_total(X):
+                X = self._add_totals(X)
+
+        # fit forecasters for each level
+        self.forecaster_ = self.forecaster.clone()
+        self.forecaster_.fit(y=y, X=X, fh=fh)
+
+        # now summation matrix
+        self.s_matrix = _get_s_matrix(y)
+
+        # parent child df
+        self.parent_child = _parent_child_df(self.s_matrix)
+
+        # bug in self.forecaster_.predict_residuals() for heir data
+        fh_resid = ForecastingHorizon(
+            y.index.get_level_values(-1).unique(), is_relative=False
+        )
+        self.residuals = y - self.forecaster_.predict(fh=fh_resid, X=X)
+
+        if self.method == "mint_cov":
+            self.g_matrix = self._get_g_matrix_mint(shrink=False)
+        elif self.method == "mint_shrink":
+            self.g_matrix = self._get_g_matrix_mint(shrink=True)
+        elif self.method == "wls_var":
+            self.g_matrix = self._get_g_matrix_mint(shrink=False, diag_only=True)
+        else:
+            raise RuntimeError("unreachable condition, error in _check_method")
+
+        return self
+
+    def _predict(self, fh, X=None):
+        """Forecast time series at future horizon.
+
+        Parameters
+        ----------
+        fh : ForecastingHorizon or None, optional (default=None)
+            The forecasting horizon with the steps ahead to to predict.
+        X : optional (default=None)
+            Exogeneous time series for the forecast
+
+        Returns
+        -------
+        y_pred : pd.Series
+            Point predictions
+        """
+        base_fc = self.forecaster_.predict(fh=fh, X=X)
+
+        if base_fc.index.nlevels < 2:
+            warn(
+                "Reconciler is intended for use with y.index.nlevels > 1. "
+                "Returning predictions unchanged."
+            )
+            return base_fc
+
+        base_fc = base_fc.groupby(level=-1)
+
+        recon_fc = []
+        for _name, group in base_fc:
+            # reconcile via SGy
+            fcst = self.s_matrix.dot(self.g_matrix.dot(group.droplevel(-1)))
+            # add back in time index
+            fcst.index = group.index
+            recon_fc.append(fcst)
+
+        recon_fc = pd.concat(recon_fc, axis=0)
+        recon_fc = recon_fc.sort_index()
+
+        return recon_fc
+
+    def _update(self, y, X=None, update_params=True):
+        """Update time series to incremental training data.
+
+        Parameters
+        ----------
+        y : pd.DataFrame
+            Time series to which to fit the forecaster.
+        X : pd.DataFrame, default=None
+            Exogenous variables based to the base forecaster
+        update_params : bool, optional (default=True)
+            whether model parameters should be updated
+
+        Returns
+        -------
+        self : reference to self
+        """
+        if y.index.nlevels < 2:
```

Comment:
the update seems incorrect or problematic.

Let's say we update only with a single time point, i.e., a length 1 `y`, then we have only a single residuals and compute matrices on only a single residuals!

Should we not use, say, `self._y` instead? In which case this would be equivalent to `fit` on all seen data, but the matrix updates could be done more efficiently, e.g., by the WSM formula. (although that might be overkill for an initial PR?)